Sources/Prometheus/MetricTypes/Histogram.swift

```diff
@@ -233,14 +233,23 @@ public class PromHistogram<NumType: DoubleRepresentable, Labels: HistogramLabels
 
     /// Helper for histograms & labels
     fileprivate func getOrCreateHistogram(with labels: Labels) -> PromHistogram<NumType, Labels> {
-        let histogram = self.subHistograms[labels]
-        if let histogram = histogram, histogram.name == self.name, histogram.help == self.help {
-            return histogram
+        let subHistograms = lock.withLock { self.subHistograms }
+        if let histogram = subHistograms[labels] {
+            if histogram.name == self.name, histogram.help == self.help {
```

Comment:
If we don't believe this can ever fail should we `assert`/`precondition` instead?